test/overloads/strict-lists/1.simple-exps

```diff
@@ -0,0 +1 @@
+[1, 1 + 1, 2 + 1, 4]
```

Comment:
How does this concrete test require narrowing?

Reply:
it's not necessarily about this test; the narrowing is needed in evaluating the `not(isKResult(Es))` side condition for heating, which is something of the sort
`isKResult(inj{KResult, KItem}(R:KResult)) => true`
which needs to fail when applied to a list of expressions which are not values.

what happens is that `inj{Exps, KItem}(ExpsCons(inj{Val, Exp}(1), ExpsCons(1 + 1, inj{Vals, Exps}(.Vals))))` must be attempted unification with `inj{KResult, KItem}(R:KResult)`. At this moment, the narrowing kicks in (`unifyOverloadingVsOverloadedVariable`), and it could narrow `R:KResult` to `inj{Vals, KResult}(ValsCons(V:Val, Vs:Vals))`, then reattempt unification between `inj{Exps, KItem}(ExpsCons(inj{Val, Exp}(1), ExpsCons(1 + 1, inj{Vals, Exps}(.Vals))))` and  `inj{Vals, KItem}(ValsCons(V:Val, Vs:Vals))`, where the latter can be transformed to `inj{Exps, KItem}(ExpCons(inj{Val,Exp}(V:Val), inj{Vals, Exps}(Vs:Vals)))`, which reduces to `V =  1` and with `ExpsCons(1 + 1, inj{Vals, Exps}(.Vals))` unified to ` inj{Vals, Exps}(Vs:Vals)`, where, applying again narrowing (this time `unifyOverloadingInjVsVariable`) we narrow `Vs:Vals` to `ValsCons(V':Val, Vs':Vals)` and use overloading to attempt unification  between `ExpsCons(1 + 1, inj{Vals, Exps}(.Vals))` and `inj{Vals, Exps}(ValsCons(V':Val, Vs':Vals))`, which gets to `1 + 1` being unified with `\inj{Val, Exp}(V':Val)`, which fails.  Maybe I should discuss this example in the documentation?

Reply:
Yes, please document this in the test semantics.